tfhe/src/integer/gpu/mod.rs

```diff
@@ -5680,9 +6267,19 @@ pub fn get_scalar_rotate_left_integer_radix_kb_size_on_gpu(
     num_blocks: u32,
     pbs_type: PBSType,
     grouping_factor: LweBskGroupingFactor,
-    noise_reduction_key: Option<&CudaModulusSwitchNoiseReductionKey>,
+    ms_configuration: Option<&CudaModulusSwitchNoiseReductionConfiguration>,
 ) -> u64 {
-    let allocate_ms_noise_array = noise_reduction_key.is_some();
+    let noise_reduction_type = ms_configuration.map_or(
```

Comment:
This is also repeated many times, could be wrapped as well in a simpler function maybe.